CHANGELOG.md

```diff
@@ -12,8 +12,157 @@ Release dates of 3.x versions:
+
+### UGens: Fixed
+
+TWindex learned to respond to multiple triggers in a single control block ([#5002](https://github.com/supercollider/supercollider/pull/5002))
+
+### sclang: Fixed
+
+MIDIOut now checks the correct bounds when sending messages ([#4652](https://github.com/supercollider/supercollider/pull/4652))
+
+WebView:-findText no longer throws an error when used correctly ([#4937](https://github.com/supercollider/supercollider/pull/4937))
+
+`[].mirror.wrapAt` no longer crashes the interpreter ([#4935](https://github.com/supercollider/supercollider/pull/4935))
+
+Fixed a segfault in the prMatchLangIP primitive ([#4927](https://github.com/supercollider/supercollider/pull/4927))
+
+Buffer:-getToFloatArray correctly handles a count of -1 now ([#4939](https://github.com/supercollider/supercollider/pull/4939))
+
+Some dead code was removed ([#4934](https://github.com/supercollider/supercollider/pull/4934))
+
+Fixed a crash on macOS when calling MIDIIn.connectAll immediately before HID.findAvailable ([#5046](https://github.com/supercollider/supercollider/pull/5046))
+
+### scel: Fixed
+
+Fixed a build issue on Ubuntu caused by incompatibility with Emacs 25 ([#4855](https://github.com/supercollider/supercollider/pull/4855))
+
+### scel: Added
+
+Added support for completion-at-point-functions and company-mode ([#4855](https://github.com/supercollider/supercollider/pull/4855))
+
+### Class library: Added
+
+Plotter learned to allow multiple active plot modes, similar to having multiple active colors ([#4459](https://github.com/supercollider/supercollider/pull/4459))
+
+The class library's HelpBrowser class learned how to respond to the same keyboard shortcuts as the IDE help browser ([#4932](https://github.com/supercollider/supercollider/pull/4932))
+
+### Class library: Changed
+
+Quarks prints more helpful error messages for a common error ([#4404](https://github.com/supercollider/supercollider/pull/4404))
+
+Object:-writeDefFile no longer allows an empty string as an argument ([#4953](https://github.com/supercollider/supercollider/pull/4953))
+
+### Class library: Fixed
+
+Cmd-W now closes GUI windows again on macOS ([#4821](https://github.com/supercollider/supercollider/pull/4821))
+
+Passing a nil spec to NamedControl no longer overwrites a spec in SynthDef metadata ([#4817](https://github.com/supercollider/supercollider/pull/4817))
+
+Fixed a deprecation that broke non-GUI builds ([#4875](https://github.com/supercollider/supercollider/pull/4875))
+
+String:-runInTerminal works again on macOS ([#4880](https://github.com/supercollider/supercollider/pull/4880))
+
+UnitTest:-assertFloatEquals's formatting was fixed ([#4946](https://github.com/supercollider/supercollider/pull/4946))
+
+String:-splitext no longer interprets a period in a path component as the start of a file extension ([#4960](https://github.com/supercollider/supercollider/pull/4960))
```

Comment:
I'm not sure anything relies on the old behaviour, but is this technically a breaking change?

Reply:
the old behavior was wrong according to spec. IMO we shouldn't label every bugfix a breaking change, or the term becomes meaningless.

Reply:
agreed - it's only breaking if user code can reasonably rely on it, which it can't in this case, just wanted to double check